ocaml/typing/uniqueness_analysis.ml

```diff
@@ -778,9 +778,19 @@ end = struct
   let child proj t = List.map (UF.Path.child proj) t
 
   let modal_child gf proj t =
-    match gf with
-    | Global_flag.Global -> untracked
-    | Global_flag.Unrestricted -> child proj t
+    (* CR zqian: Instead of just ignoring such children, we should add modality
+       to [Projection.t] and add corresponding logic in [UsageTree]. *)
+    let gf = Modality.Value.to_list gf in
+    let l =
+      List.filter
+        (function
+         | Atom (Monadic Uniqueness, Join_with Shared) -> true
+         | Atom (Comonadic Linearity, Meet_with Many) -> true
+         | _ -> false
+          : Modality.t -> _)
+        gf
+    in
+    if List.length l = 2 then untracked else child proj t
```

Comment:
As discussed, we will let this slip, because the uniqueness_analysis is not used in practice yet; and the code here is a conservative approximation of the proper behaviour.